cpp/src/arrow/compute/exec/exec_plan.cc

```diff
@@ -350,37 +350,38 @@ void MapNode::StopProducing() {
 
 Future<> MapNode::finished() { return finished_; }
 
-void MapNode::SubmitTask(std::function<Result<ExecBatch>(ExecBatch)> map_fn,
-                         ExecBatch batch) {
+void MapNode::SubmitTask(std::function<Result<ExecBatch>()> map_fn) {
```

Comment:
I don't know if `MapNode` will still need to exist.  Eventually this logic will be moving over to the scheduler I think.

Reply:
Indeed, MapNode should go away once we have the scheduler in place, this PR is not for doing changes in this part of the logic, but just to accommodate the ExecNode interface to make it ready for future PRs about the scheduler.